File: scripts/visualize_data.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import os

# Create outputs directory if it doesn't exist
os.makedirs('../outputs', exist_ok=True)

# Read the data
df = pd.read_csv('../data/customer_sales.csv')
df['date'] = pd.to_datetime(df['date'])

# Calculate daily sales
daily_sales = df.groupby('date')['total_spent'].sum().reset_index()

# Create daily sales trend plot
plt.figure(figsize=(12, 6))
sns.lineplot(data=daily_sales, x='date', y='total_spent', marker='o')
plt.title('Daily Sales Trend')
plt.xlabel('Date')
plt.ylabel('Total Sales ($)')
plt.xticks(rotation=45)
plt.tight_layout()
plt.savefig('../outputs/daily_sales_trend.png')
plt.close()

# Create age distribution plot
plt.figure(figsize=(10, 6))
sns.histplot(data=df, x='age', bins=5, hue='gender', multiple='stack')
plt.title('Age Distribution by Gender')
plt.xlabel('Age')
plt.ylabel('Count')
plt.tight_layout()
plt.savefig('../outputs/age_distribution.png')
plt.close()

# Calculate and save average spending by age group
df['age_group'] = pd.cut(df['age'], bins=[20, 30, 40, 50, 60], labels=['20-30', '31-40', '41-50', '51-60'])
age_group_stats = df.groupby('age_group')['total_spent'].agg(['mean', 'count'])
age_group_stats.to_csv('../outputs/age_group_statistics.csv') 

File: scripts/visualize_data.R
# Load required packages
library(tidyverse)
library(ggplot2)
library(dplyr)
library(readr)
library(lubridate)

# Create outputs directory if it doesn't exist
dir.create("../outputs", showWarnings = FALSE)

# Read the data
df <- read_csv("../data/customer_sales.csv") %>%
  mutate(date = as.Date(date))

# Calculate daily sales
daily_sales <- df %>%
  group_by(date) %>%
  summarise(total_sales = sum(total_spent))

# Create daily sales trend plot
ggplot(daily_sales, aes(x = date, y = total_sales)) +
  geom_line() +
  geom_point() +
  labs(title = "Daily Sales Trend",
       x = "Date",
       y = "Total Sales ($)") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        plot.title = element_text(hjust = 0.5))
ggsave("../outputs/daily_sales_trend.png", width = 12, height = 6)

# Create age distribution plot
ggplot(df, aes(x = age, fill = gender)) +
  geom_histogram(bins = 5, position = "stack") +
  labs(title = "Age Distribution by Gender",
       x = "Age",
       y = "Count") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
ggsave("../outputs/age_distribution.png", width = 10, height = 6)

# Calculate and save average spending by age group
age_group_stats <- df %>%
  mutate(age_group = cut(age, 
                        breaks = c(20, 30, 40, 50, 60),
                        labels = c("20-30", "31-40", "41-50", "51-60"))) %>%
  group_by(age_group) %>%
  summarise(
    mean_spent = mean(total_spent),
    count = n()
  )

write_csv(age_group_stats, "../outputs/age_group_statistics.csv") 